test/SparqlAntlrParserTest.cpp

```diff
@@ -1053,36 +1043,85 @@ TEST(SparqlParser, propertyListPathNotEmpty) {
-         const std::vector<ad_utility::sparql_types::PathTriple>& triples) {
-        expectCompleteParse(parseTriplesSameSubjectPath(input, {}),
-                            IsTriplesSameSubjectPath(triples));
+         const std::vector<ad_utility::sparql_types::TripleWithPropertyPath>&
+             triples) {
+        expectCompleteParse(parseTriplesSameSubjectPath(input),
+                            testing::Eq(triples));
       };
-  expectTriplesSameSubjectPath(
-      "?foo <bar> ?baz",
-      {{Variable{"?foo"}, PropertyPath::fromIri("<bar>"), Variable{"?baz"}}});
-  expectTriplesSameSubjectPath(
-      "?foo <bar> ?baz ; <mehr> ?t",
-      {{Variable{"?foo"}, PropertyPath::fromIri("<bar>"), Variable{"?baz"}},
-       {Variable{"?foo"}, PropertyPath::fromIri("<mehr>"), Variable{"?t"}}});
-  expectTriplesSameSubjectPath(
-      "?foo <bar> ?baz , ?t",
-      {{Variable{"?foo"}, PropertyPath::fromIri("<bar>"), Variable{"?baz"}},
-       {Variable{"?foo"}, PropertyPath::fromIri("<bar>"), Variable{"?t"}}});
-  expectTriplesSameSubjectPath(
-      "?foo <bar> ?baz , ?t ; <mehr> ?d",
-      {{Variable{"?foo"}, PropertyPath::fromIri("<bar>"), Variable{"?baz"}},
-       {Variable{"?foo"}, PropertyPath::fromIri("<bar>"), Variable{"?t"}},
-       {Variable{"?foo"}, PropertyPath::fromIri("<mehr>"), Variable{"?d"}}});
-  expectTriplesSameSubjectPath(
-      "?foo <bar> ?baz ; <mehr> ?t , ?d",
-      {{Variable{"?foo"}, PropertyPath::fromIri("<bar>"), Variable{"?baz"}},
-       {Variable{"?foo"}, PropertyPath::fromIri("<mehr>"), Variable{"?t"}},
-       {Variable{"?foo"}, PropertyPath::fromIri("<mehr>"), Variable{"?d"}}});
-}
\ No newline at end of file
+  auto PathIri = &PropertyPath::fromIri;
+  using Var = Variable;
+  expectTriples("?foo <bar> ?baz",
+                {{Var{"?foo"}, PathIri("<bar>"), Var{"?baz"}}});
+  expectTriples("?foo <bar> ?baz ; <mehr> ?t",
+                {{Var{"?foo"}, PathIri("<bar>"), Var{"?baz"}},
+                 {Var{"?foo"}, PathIri("<mehr>"), Var{"?t"}}});
+  expectTriples("?foo <bar> ?baz , ?t",
+                {{Var{"?foo"}, PathIri("<bar>"), Var{"?baz"}},
+                 {Var{"?foo"}, PathIri("<bar>"), Var{"?t"}}});
+  expectTriples("?foo <bar> ?baz , ?t ; <mehr> ?d",
+                {{Var{"?foo"}, PathIri("<bar>"), Var{"?baz"}},
+                 {Var{"?foo"}, PathIri("<bar>"), Var{"?t"}},
+                 {Var{"?foo"}, PathIri("<mehr>"), Var{"?d"}}});
+  expectTriples("?foo <bar> ?baz ; <mehr> ?t , ?d",
+                {{Var{"?foo"}, PathIri("<bar>"), Var{"?baz"}},
+                 {Var{"?foo"}, PathIri("<mehr>"), Var{"?t"}},
+                 {Var{"?foo"}, PathIri("<mehr>"), Var{"?d"}}});
+  expectTriples("<foo> <bar> ?baz ; ?mehr \"a\"",
+                {{Iri("<foo>"), PathIri("<bar>"), Var{"?baz"}},
+                 {Iri("<foo>"), Var("?mehr"), Literal("\"a\"")}});
+  expectTriples("_:1 <bar> ?baz",
+                {{BlankNode(false, "1"), PathIri("<bar>"), Var{"?baz"}}});
+}
```

Comment:
Do we somewhere already have tests for the cornercases (numeric literals, the ql_contains_word prediate, other shenanigans?